### PR TITLE
feat: secondary activation flow

### PR DESCRIPTION
**Pull request by kappa90**
## Problem
See https://github.com/PostHog/meta/pull/65

We want to push users to complete secondary actions after the main ingestion flow

## Changes
- Add a secondary sidebar that is activated after the main ingestion flow
- Users can complete secondary tasks, for now:
    - Ingest an event: redirects to the main ingestion flow
    - Invite a team member: opens the invitation modal
    - Setup session recordings: shows the session recordings settings in the sidebar
    - Track custom actions: redirects them to the right doc (will improve this in a second iteration so they don't have to leave the app), and tracks if there's any custom event definition
    - Install an app: sends them to the apps marketplace
- Tasks can be skipped (list of skipped tasks is saved locally in local storage, which means logging in from a different device will show again the skipped tasks as uncompleted, good enough for v1)
- The sidebar reminder on the top bar disappears once all tasks are completed or skipped
- This sidebar can always be found again from the Help Button
- Substitutes the prompts based activation flow, which had a CTA to track marketing websites, in a second iteration we'll include a way to open the docs in the sidebar, and show additional non-blocking tasks such as this one
- Bonus: there's a cute professor hedgehog

## Design
I put the sidebar on the right as the small button in the topbar couldn't be fit anywhere else.
 
image

image

## Copy
This is the buttons copy (tbd):
```
- Ingest your first event
     - Ingest your first event to get started with PostHog
- Invite a team member
     - Every person in your company can benefit from PostHog
- Setup session recordings
     - See how your users are using your product
- Track custom events
     - Track custom events to get more insights into your product
- Install your first app
     - Extend PostHog's core functionality with apps
```

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?
- Added a small logic test

### CODE REVIEW

**Comment by kappa90** [comment]
@mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?

**Comment by alexkim205** [comment]
> @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?

Have you tried to regenerate the type files by running `yarn typegen:clean`?

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { useActions, useValues } from 'kea'
+import { navigationLogic } from '~/layout/navigation/navigationLogic'
+import { activationLogic, Task } from './activationLogic'
+import './ActivationSidebar.scss'
+import { Progress } from 'antd'
+import { IconCheckmark, IconClose } from '../icons'
+import { SessionRecording as SessionRecordingConfig } from 'scenes/project/Settings/SessionRecording'
+import { ProfessorHog } from '../hedgehogs'
+
+const Task = ({ id, name, description, completed, canSkip, skipped, url }: Task): JSX.Element => {
```
Comment (alexkim205) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:12]: Rename `Task` to `TaskProps` or something different so that we don't redefine Task again

File: frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+import { inviteLogic } from 'scenes/organization/Settings/inviteLogic'
+import { membersLogic } from 'scenes/organization/Settings/membersLogic'
+import { pluginsLogic } from 'scenes/plugins/pluginsLogic'
+import { teamLogic } from 'scenes/teamLogic'
+import { navigationLogic } from '~/layout/navigation/navigationLogic'
+import { EventDefinitionType } from '~/types'
+
+import type { activationLogicType } from './activationLogicType'
+
+export enum ACTIVATION_TASKS {
```
Comment (alexkim205) [inline frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts:14]: Nit: we usually don't capitalize enums, and should probably follow the standard here as well

File: frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                                completed: currentTeam?.ingested_event ?? false,
+                                canSkip: false,
+                                skipped: false,
+                            })
+                            break
+                        case ACTIVATION_TASKS.INVITE_TEAM_MEMBER:
+                            tasks.push({
+                                id: ACTIVATION_TASKS.INVITE_TEAM_MEMBER,
+                                name: 'Invite a team member',
+                                description: 'Every person in your company can benefit from PostHog',
```
Comment (joethreepwood) [inline frontend/src/lib/components/ActivationSidebar/ActivationLogic.ts:129]: Users may not be in a company. They could be in some other sort of organization. We could say team? But then that may be too limiting.

```suggestion
                                description: 'Everyone in your organization can benefit from PostHog',
```

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                            }
+                        }}
+                    />
+                </div>
+                {showSessionRecordingConfig ? (
+                    <SessionRecordingConfig />
+                ) : (
+                    <>
+                        <h2 className="subtitle">Quick Start</h2>
+                        <p>Learn how to get the most out of PostHog, straigth away.</p>
```
Comment (joethreepwood) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:84]:
```suggestion
                        <p>Learn how to get the most out of PostHog, straight away.</p>
```

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                        <div className="my-4 flex items-center justify-center">
+                            <div className="flex flex-col items-center">
+                                <Progress
+                                    type="circle"
+                                    strokeWidth={10}
+                                    percent={completionPercent}
+                                    format={() => activeTasks.length}
+                                />
+                                <p className="text-muted mt-2">Remaining tasks</p>
```
Comment (joethreepwood) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:93]: It may be worth taking the word 'task' out, just so it doesn't feel so choreish?

"Still to go" maybe?

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                            </div>
+                            <div className="ActivationSideBar__hog">
+                                <ProfessorHog className="max-h-full w-auto object-contain" />
+                            </div>
+                        </div>
+                        {activeTasks.length > 0 && (
+                            <div className="mt-4">
+                                <div className="text-muted uppercase text-xs">Next tasks</div>
```
Comment (joethreepwood) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:101]: In keeping with tasks being choreish, maybe: "What's next?"

File: frontend/src/lib/components/HelpButton/HelpButton.tsx [hunk 7]
```diff
@@ -145,19 +147,17 @@ export function HelpButton({
...
+                        <LemonButton
+                            icon={<IconTrendingUp />}
+                            status="stealth"
+                            fullWidth
+                            onClick={() => {
+                                toggleActivationSideBar()
+                                hideHelp()
+                            }}
+                        >
+                            Quick Start using PostHog
```
Comment (joethreepwood) [inline frontend/src/lib/components/HelpButton/HelpButton.tsx:159]:
```suggestion
                            Quick Start
```
'Using' just sounds weird here to me. And it's a little redundant.

**Review by joethreepwood — APPROVED** [review]
Some language tweaks, but lgtm.

**Comment by kappa90** [comment]
- Fixed the copy and code based on @alexkim205 and @joethreepwood's reviews (thank you!)
- Added a tracking event when this sidebar is opened showing how many tasks have been completed
- Fixed the reducer for skipped tasks to be per-project, rather than per-user

**Comment by kappa90** [comment]
> > @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?
> 
> Have you tried to regenerate the type files by running `yarn typegen:clean`?

That didn't seem to work 😢

**Comment by alexkim205** [comment]
> > > @mariusandra I renamed a logic from `ActivationLogic` to `activationLogic` but the old logic name is still picked up and the tests fail, do you know what's going on?
> > 
> > 
> > Have you tried to regenerate the type files by running `yarn typegen:clean`?
> 
> That didn't seem to work 😢

Pushed some changes that should fix this (I hope)! Also refactored things in the logics a bit to remove dead and buggy code.

**Review by alexkim205 — APPROVED** [review]

File: frontend/src/lib/components/ActivationSidebar/activationLogic.ts [hunk 8]
```diff
@@ -245,20 +248,10 @@ export const activationLogic = kea<activationLogicType>([
...
-        [router.actionTypes.locationChanged]: async () => {
-            actions.hideActivationSideBar()
-        },
     })),
     events(({ actions }) => ({
         afterMount: () => {
             actions.loadCustomEvents()
         },
     })),
-    urlToAction(({ actions, values }) => ({
```
Comment (kappa90) [inline frontend/src/lib/components/ActivationSidebar/activationLogic.ts]: Why did you remove this?

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+import { ProfessorHog } from '../hedgehogs'
+
+const Task = ({ id, name, description, completed, canSkip, skipped, url }: ActivationTaskType): JSX.Element => {
+    const displaySideAction = !completed && !skipped && canSkip
+    const { runTask, skipTask } = useActions(activationLogic)
+
+    const content = (
+        <div className="my-4 mx-2">
+            <p className="m-0">{name}</p>
+            {!completed && !skipped && <p className="font-normal text-xs mt-2 mb-0">{description}</p>}
```
Comment (raquelmsmith) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:19]: For whatever reason all `<p>` tags have a 5px left margin. We should remove it here since it looks oddly indented.

image

File: frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+                            }
+                        }}
+                    />
+                </div>
+                {showSessionRecordingConfig ? (
+                    <SessionRecordingConfig />
+                ) : (
+                    <>
+                        <h2 className="subtitle">Quick Start</h2>
+                        <p>Learn how to get the most out of PostHog, straight away.</p>
```
Comment (raquelmsmith) [inline frontend/src/lib/components/ActivationSidebar/ActivationSidebar.tsx:84]: The `, straight away.` reads weird to me, and I (as a user) am not sure what it means to "get the most out of PostHog". I'd reword to something like...

```suggestion
                        <p>Use our Quick Start guide to learn about everything PostHog can do for you and your product.</p>
```

File: frontend/src/layout/navigation/navigationLogic.ts [hunk 11]
```diff
@@ -23,6 +23,9 @@ export const navigationLogic = kea<navigationLogicType>({
     actions: {
         toggleSideBarBase: true,
         toggleSideBarMobile: true,
+        toggleActivationSideBar: true,
+        showActivationSideBar: true,
+        hideActivationSideBar: true,
```
Comment (raquelmsmith) [inline frontend/src/layout/navigation/navigationLogic.ts:28]: It seems a little unnecessary to have a `toggle` plus a `hide` and a `show`. Theoretically we can just do with a `toggle`, yes?

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 12]
```diff
@@ -480,6 +480,11 @@ export const eventUsageLogic = kea<eventUsageLogicType>({
         reportInstanceSettingChange: (name: string, value: string | boolean | number) => ({ name, value }),
         reportAxisUnitsChanged: (properties: Record<string, any>) => ({ ...properties }),
         reportTeamSettingChange: (name: string, value: any) => ({ name, value }),
+        reportActivationSideBarShown: (
```
Comment (raquelmsmith) [inline frontend/src/lib/utils/eventUsageLogic.ts:483]: I think it would also be good to record events for when people click the different actions within the sidebar so we can track how well the sidebar works for getting people to do the different things.

File: frontend/src/lib/components/ActivationSidebar/activationLogic.ts [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+import { inviteLogic } from 'scenes/organization/Settings/inviteLogic'
+import { membersLogic } from 'scenes/organization/Settings/membersLogic'
+import { pluginsLogic } from 'scenes/plugins/pluginsLogic'
+import { teamLogic } from 'scenes/teamLogic'
+import { navigationLogic } from '~/layout/navigation/navigationLogic'
+import { EventDefinitionType, TeamBasicType } from '~/types'
+import type { activationLogicType } from './activationLogicType'
+import { urls } from 'scenes/urls'
+
+export enum ActivationTasks {
```
Comment (raquelmsmith) [inline frontend/src/lib/components/ActivationSidebar/activationLogic.ts:15]: We should def add tasks for creating a dashboard, viewing an insight, etc. Most people are signing up for product analytics and we want to make sure they are actually getting a good tour of that product before we push them onto something else like recordings, esp since our activation metric depends on that.

**Review by raquelmsmith — CHANGES_REQUESTED** [review]
Love it! Few comments, but overall excited to see how this works. 

Have we decided how we are planning to define success/failure of this?

File: frontend/src/layout/navigation/navigationLogic.ts (hunk 11 above)
Comment (kappa90) [reply frontend/src/layout/navigation/navigationLogic.ts:28]: There's certain places where I always want to either show or hide, so it's the toggle that's just an extra, I'd just keep them, since it's just kea actions

File: frontend/src/lib/utils/eventUsageLogic.ts (hunk 12 above)
Comment (kappa90) [reply frontend/src/lib/utils/eventUsageLogic.ts:483]: Makes sense, done!

File: frontend/src/lib/components/ActivationSidebar/activationLogic.ts (hunk 13 above)
Comment (kappa90) [reply frontend/src/lib/components/ActivationSidebar/activationLogic.ts:17]: Done done done

**Comment by kappa90** [comment]
- Added two more tasks: `Create your first insight` and `Create your first dashboard`
- Added a `activation sidebar task clicked` event
- Improved the UI based on @raquelmsmith's feedback

**Review by raquelmsmith — APPROVED** [review]
:shipit:

**Comment by kappa90** [comment]
Regarding considering this a success or not, we should check if individual CTA push to an increase in the related metric (discoveries, plugins installed, session recording on, etc.)
